src/irr/asset/CGLTFLoader.cpp

```diff
@@ -34,16 +35,427 @@ namespace irr
+						}
+						else
+							continue;
+					}
+				}
+
+				/*
+					Each sampler specifies filter and wrapping options corresponding to the GL types.
+				*/
+
+				else if (key == "samplers")
+				{
+					for (auto& sampler : value)
+					{
+						auto& magFilter = sampler.at_key("magFilter");
+						if (magFilter.error() != simdjson::error_code::NO_SUCH_FIELD)
+							magFilter.get_uint64().value(); // TODO
+
+						auto& minFilter = sampler.at_key("minFilter");
+						if (minFilter.error() != simdjson::error_code::NO_SUCH_FIELD)
+							minFilter.get_uint64().value(); // TODO
+
+						auto& wrapS = sampler.at_key("wrapS");
+						if (wrapS.error() != simdjson::error_code::NO_SUCH_FIELD)
+							wrapS.get_uint64().value(); // TODO
+
+						auto& wrapT = sampler.at_key("wrapT");
+						if (wrapT.error() != simdjson::error_code::NO_SUCH_FIELD)
+							wrapT.get_uint64().value(); // TODO
+					}
+				}
+
+				/*
+					There are materials using a common set of parameters that are based on widely 
+					used material representations from Physically-Based Rendering (PBR).
+				*/
+
+				else if (key == "materials")
+				{
+					// TODO
+				}
+
+				/*
+					A camera defines the projection matrix that transforms from view to clip coordinates.
+				*/
+
+				else if (key == "cameras")
+				{
+					for (auto& camera : value)
+					{
+						auto& type = camera.at_key("type");
+						if (type.error() == simdjson::error_code::NO_SUCH_FIELD)
+							continue;
+						else
+						{
+							auto& typeVal = type.get_string().value();
+
+							if (typeVal == "perspective")
+							{
+								auto& perspective = camera.at_key("perspective");
+								if (perspective.error() == simdjson::error_code::NO_SUCH_FIELD)
+									continue;
+
+								auto& yfov = perspective.at_key("yfov");
+								if (yfov.error() != simdjson::error_code::NO_SUCH_FIELD)
+								{
+									auto& yfovVal = yfov.get_double().value();
+									if (yfovVal <= 0)
+										continue;
+								}
+								else
+									continue;
+
+								auto& znear = perspective.at_key("znear");
+								if (znear.error() != simdjson::error_code::NO_SUCH_FIELD)
+								{
+									auto& znearVal = znear.get_double().value();
+									if (znearVal <= 0)
+										continue;
+								}
+								else
+									continue;
+
+							}
+							else if (typeVal == "orthographic")
+							{
+								auto& orthographic = camera.at_key("orthographic");
+								if (orthographic.error() == simdjson::error_code::NO_SUCH_FIELD)
+									continue;
+
+								auto& xmag = orthographic.at_key("xmag");
+								if (xmag.error() != simdjson::error_code::NO_SUCH_FIELD)
+								{
+									auto& xmagVal = xmag.get_double().value();
+								}
+								else
+									continue;
+
+								auto& ymag = orthographic.at_key("ymag");
+								if (ymag.error() != simdjson::error_code::NO_SUCH_FIELD)
+								{
+									auto& ymagVal = ymag.get_double().value();
+								}
+								else
```

Comment:
all this scene and camera stuff needs to go into metadata.

Reply:
it will go there then